portfolio/src/main/webapp/script.js

```diff
@@ -102,20 +102,119 @@ function getComments() {
 class Countries {
 
   constructor(jsonObj) {
-    this.keyedJSON = jsonObj;
+
+    //  = JSON.parse(JSON.stringify(jsonObj));
+    this.keyedJSONBackup = JSON.parse(JSON.stringify(jsonObj));
+    this.keyedJSON = this.keyedJSONBackup;
+
+    this.naJSON = this.filterJsonByContinent("NA");
+    this.saJSON = this.filterJsonByContinent("SA");
+    this.euJSON = this.filterJsonByContinent("EU");
+    this.afJSON = this.filterJsonByContinent("AF");
+    this.asJSON = this.filterJsonByContinent("AS");
+    this.ocJSON = this.filterJsonByContinent("OC");
+
+    this.naBool = true;
+    this.saBool = true;
+    this.euBool = true;
+    this.afBool = true;
+    this.asBool = true;
+    this.ocBool = true;
   }
 
   getRandCountryCode() {
     return Object.keys(this.keyedJSON)[Math.floor(Math.random() * Object.keys(this.keyedJSON).length)];
   }
 
   removeCountry(countryCode) {
+
+    const continentStr = this.keyedJSON[countryCode].continent;
+
     delete this.keyedJSON[countryCode];
+    delete this.keyedJSONBackup[countryCode];
+
+    if (continentStr === "NA") {
+      delete this.naJSON[countryCode];
+    }
+    else if (continentStr === "SA") {
+      delete this.saJSON[countryCode];
+    }
+    else if (continentStr === "EU") {
+      delete this.euJSON[countryCode];
+    }
+    else if (continentStr === "AF") {
+      delete this.afJSON[countryCode];
+    }
+    else if (continentStr === "AS") {
+      delete this.asJSON[countryCode];
+    }
+    else {
+      delete this.ocJSON[countryCode];
+    }
   }
 
   empty() {
     return Object.keys(this.keyedJSON).length === 0;
   }
+
+  filterJsonByContinent(continentStr) {
```

Comment:
Is there a way in javascript to make functions and variables private? The function ```filterJsonByContinent``` doesn't really need to be public.

Reply:
Yeah, it doesn't look like you can in a Javascript class (https://www.sitepoint.com/javascript-private-class-fields/).